server/views/partials/_calendar.njk

```diff
@@ -0,0 +1,25 @@
+{% macro occupancyCalendar(calendar) %}
+    <div class="calendar">
+        {% for month in calendar %}
+            <section>
+                <h3 class="govuk-heading-m">{{ month.name }}</h3>
+
+                <ul class="calendar__month govuk-list">
+                    {% for day in month.days %}
+                        <li class="calendar__day calendar__day--{{ day.name.slice(0,3) | lower }}{% if day.status !== 'available' %} {{ 'govuk-tag--yellow' if day.status in ['availableForCriteria','full'] else 'govuk-tag--red' }}{% endif %}">
+                            <a class="calendar__link" href="{% if day.link %}{{ day.link }}{% else %}#{% endif %}">
```

Comment:
So this assumes that all calendars are rendered with a `link` property in their day structures. I could put this in the lower level day templates, but that would mean more of the day rendering, including the date name, would be duplicated.